CONTRACT_TESTING.md

```diff
@@ -0,0 +1,65 @@
+
+For that reason, we are going to use Pact to get notified whenever the contracts are not satisfied. The steps that we are folowing are:
+
+1. Create unit tests mocking the HTTP requests/responses for Fleet's APIs. As we all know, this approach will always work, as we predefine the mocks, but on the other hand we won't get notified if/when the provider changes, so we must be reactively checking for updates.
+1. Create a contract/pact for the consumer (the e2e tests), with the behaviours we expect from Fleet. The result will be a JSON file with the specs.
+1. Store the contract in some place that is accesible by the provider. In this case, we are using the file system to store them, but Pact recommends using a [Broker](https://github.com/pact-foundation/pact_broker).
+1. Verify the contract/pact from the provider side. The provider is able to check if a change breaks all its consumers by itself, as it has access to the contracts they provided (via file system or the afore mentioned Broker).
+
+>Ideally, the provider project should be the one responsible of the verification but, for simplification, we are creating that verification in this project. Kibana should implement a way to verify contracts/pacts from its own build system, so any engineer is able to check if he/she breaks the consumers.
+
+### Executing the tests
+From root directory, use the following Make targets:
+
+#### Generate the contracts for the consumer
+```shell
+$ make pact-consumer
+```
+
+It will execute Pact (using the [Go implementation](https://github.com/pact-foundation/pact-go/)) to generate the contract for the consumer (the E2E tests), generating a [JSON file](./pacts/e2e_testing_framework-fleet.json) representing the contract. This file will contain all the scenarios defined in the contract, highlighting the following sections:
+
+- **consumer**: the name of the consumer
+- **provider**: the name of the provider
+- **interactions**: the scenarios executed in the contract, defining the format of the HTTP request, HTTP headers and HTTP response for each one. It will use regex to handle the response values, so that it's not coupled with the real response data returned by the provider.
+- **meta**: Pact specification and version (2.0.0)
+
+
+#### Verify the contracts from provider side
+We want to verify that the provider satisfies the contracts. For that reason, we need the provider to be started first, and to achieve it, we are providing the following Make commands:
+
+```shell
+$ make prepare-pact-provider-deps # will run an Elasticsearch, Kibana andd Package Registry
+$ make pact-provider              # will run the verification of the contracts by itself
+$ make verify-provider            # will tear down the runtime dependencies
```

Comment:
`verify-provider` seems to call some other make goals -> https://github.com/elastic/e2e-testing/pull/339/files#diff-b67911656ef5d18c4ae36cb6741b7965R47

Is the comment correct?

Reply:
Yes, it's on purpose: verify is the last piece in the chain, defining a well-known life cycle: prepare > verify > destroy

I did this that way to support decoupling the destroy goal in the post stage of the CI pipeline (or to be run manually)